scripts/convert_merged_tract_to_dpdd.py

```diff
@@ -158,17 +167,24 @@ def write_dataframe_to_files(
     if verbose:
         print("Writing {} {} to HDF5 DPDD file.".format(tract, patch))
     key = key_format.format(**info)
-    df.to_hdf(outfile_base_tract + '.hdf5', key=key)
+    hdf_file = outfile_base_tract+'.hdf5'
+    # Append iff the file already exists
+    hdf_append = append and os.path.exists(hdf_file)
+    df.to_hdf(hdf_file, key=key, append=hdf_append, format='table')
 
-    # In principle, Parquet should be really happy
-    # with files appended to HDFS-type storage
-    # But the Pandas implementation doesn't expose this ability.
     if verbose:
         print("Writing {} {} to Parquet DPDD file.".format(tract, patch))
-    df.to_parquet(
-        outfile_base_tract_patch + '.parquet',
-        engine=parquet_engine,
-        compression=parquet_compression)
+    parquet_file = outfile_base_tract+'.parquet'
+    # Append iff the file already exists
+    parquet_append = append and os.path.exists(parquet_file)
+    df.to_parquet(parquet_file,
```

Comment:
### pyarrow
The options `append` and `file_scheme` are not recognized when using `pyarrow` engine:
```
$ python convert_merged_tract_to_dpdd.py --tract 4432 --reader dc2_object_run1.1p_tract4432 --parquet_engine pyarrow
Writing 4432 05 to HDF5 DPDD file.
Writing 4432 05 to Parquet DPDD file.
Traceback (most recent call last):
  File "convert_merged_tract_to_dpdd.py", line 258, in <module>
    parquet_compression=args.parquet_compression)
  File "convert_merged_tract_to_dpdd.py", line 83, in convert_tract_to_dpdd
    convert_cat_to_dpdd(cat, **kwargs)
  File "convert_merged_tract_to_dpdd.py", line 110, in convert_cat_to_dpdd
    write_dataframe_to_files(quantities_this_patch, **kwargs)
  File "convert_merged_tract_to_dpdd.py", line 184, in write_dataframe_to_files
    compression=parquet_compression)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py", line 1945, in to_parquet
    compression=compression, **kwargs)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/io/parquet.py", line 257, in to_parquet
    return impl.write(df, path, compression=compression, **kwargs)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/io/parquet.py", line 121, in write
    coerce_timestamps=coerce_timestamps, **kwargs)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pyarrow/parquet.py", line 1092, in write_table
    **kwargs) as writer:
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pyarrow/parquet.py", line 311, in __init__
    **options)
  File "pyarrow/_parquet.pyx", line 810, in pyarrow._parquet.ParquetWriter.__cinit__
TypeError: __cinit__() got an unexpected keyword argument 'append'
```

 I guess this is a missing feature from the pandas API.

### fastparquet
The script seems to run ok until the very end when I get:
```
$ python convert_merged_tract_to_dpdd.py --tract 4432 --reader dc2_object_run1.1p_tract4432 --parquet_engine fastparquet
...
Writing 4432 61 to HDF5 DPDD file.
Writing 4432 61 to Parquet DPDD file.
Traceback (most recent call last):
  File "convert_merged_tract_to_dpdd.py", line 258, in <module>
    parquet_compression=args.parquet_compression)
  File "convert_merged_tract_to_dpdd.py", line 83, in convert_tract_to_dpdd
    convert_cat_to_dpdd(cat, **kwargs)
  File "convert_merged_tract_to_dpdd.py", line 110, in convert_cat_to_dpdd
    write_dataframe_to_files(quantities_this_patch, **kwargs)
  File "convert_merged_tract_to_dpdd.py", line 184, in write_dataframe_to_files
    compression=parquet_compression)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py", line 1945, in to_parquet
    compression=compression, **kwargs)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/io/parquet.py", line 257, in to_parquet
    return impl.write(df, path, compression=compression, **kwargs)
  File "/Users/julien/anaconda3/lib/python3.6/site-packages/pandas/io/parquet.py", line 218, in write
    compression=compression, **kwargs)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 879, in write
    compression=compression, fmd=fmd)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 626, in make_part_file
    rg = make_row_group(f, data, schema, compression=compression)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 614, in make_row_group
    compression=comp)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 513, in write_column
    data, selement)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 252, in encode_plain
    out = convert(data, se)
  File "/Users/julien/.local/lib/python3.6/site-packages/fastparquet/writer.py", line 167, in convert
    out = np.packbits(padded.reshape(-1, 8)[:, ::-1].ravel())
TypeError: Expected an input array of integer or boolean data type
```
If I remove the option `append`, the process is fine however.

Reply:
It works fine for me:
`python convert_merged_tract_to_dpdd.py --tract 4432 --parquet_engine pyarrow`

What's in your custom `dc2_object_run1.1p_tract4432` reader?

Reply:
I ran with
```
python 3.6
pyarrow 0.10.0
pandas 0.23.4
```

Reply:
Following on this, I see that the script still crashes with pyarrow (see links above). Here is a simple example to reproduce the error outside of this script if needed (ran on JupyerLab at nersc using the `desc-python` kernel + pip install --user for fastparquet & pyarrow):

```python
import pyarrow
import pandas as pd
print(pyarrow.__version__) # 0.11.1
print(pd.__version__) # 0.23.4

data = {'col_1': [3, 2, 1, 0], 'col_2': ['a', 'b', 'c', 'd']}
df = pd.DataFrame.from_dict(data)

# Fastparquet goes fine
df.to_parquet('fastparquet_s.parquet', compression=None, engine="fastparquet", file_scheme="simple")
df.to_parquet('fastparquet_h.parquet', compression=None, engine="fastparquet", file_scheme="hive")

# Pyarrow complains about `file_scheme` unexpected
df.to_parquet('pyarrow_s.parquet', compression=None, engine="pyarrow", file_scheme="simple")
df.to_parquet('pyarrow_h.parquet', compression=None, engine="pyarrow", file_scheme="hive")
```

The reason is that `pd.to_parquet` with `pyarrow` engine is a wrapper around `pyarrow.parquet.write_table`.